codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java

```diff
@@ -461,7 +464,7 @@ private void parseBodyAttributesStandard() {
                     } else if (read == HttpConstants.LF) {
                         currentStatus = MultiPartStatus.PREEPILOGUE;
                         ampersandpos = currentpos - 1;
-                        setFinalBuffer(undecodedChunk.retainedSlice(firstpos, ampersandpos - firstpos));
+                        setFinalBuffer(undecodedChunk.copy(firstpos, ampersandpos - firstpos));
```

Comment:
hmm... retainedSlice(...) should be fine. This will do an extra memory copy which I think is not needed.